java/android/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.java

```diff
@@ -0,0 +1,704 @@
+        API_KEY.put(Environment.PRODUCTION, "");
+    }
+
+    private final Plan observePlan = new Plan()
+        .setBranch("main")
+        .setSource("java-ampli")
+        .setVersion("0");
+
+    private AmplitudeClient client;
+
+    public AmplitudeClient getClient() {
+        return this.client;
+    }
+
+    private boolean disabled = false;
+
+    public void load(LoadOptions options) {
+        Boolean disabled = options.getDisabled();
+        this.disabled = disabled != null ? disabled : false;
+        if (this.client != null) {
+            System.err.println("Warning: Ampli is already initialized. Ampli.getInstance().load() should be called once at application start up.");
+            return;
+        }
+
+        Environment env = options.getEnvironment();
+        if (env == null) {
+            env = Environment.DEVELOPMENT;
+        }
+
+        LoadClientOptions clientOptions = options.getClient();
+
+        String apiKey = Ampli.API_KEY.get(env);
+        if (clientOptions != null) {
+            String optionsApiKey = clientOptions.getApiKey();
+            if (optionsApiKey != null) {
+                apiKey = optionsApiKey;
+            }
+        }
+
+        if (clientOptions != null) {
+            AmplitudeClient client = clientOptions.getInstance();
+            if (client != null) {
+                this.client = client;
+            }
+        }
+
+        if (this.client == null) {
+            this.client = Amplitude.getInstance();
+            this.client.initialize(options.getAndroidContext(), apiKey);
+        } else {
+            System.err.println("Ampli.getInstance().load() requires 'environment', 'client.apiKey', or 'client.instance'");
+            return;
+        }
+
+        Plan plan = observePlan;
+        if (clientOptions != null) {
+            Plan optionsPlan = clientOptions.getPlan();
+            if (optionsPlan != null) {
+                plan = optionsPlan;
+            }
+        }
+        this.client.setPlan(plan);
+    }
+
+    public void track(Event event) {
+        this.track(event, null, null);
+    }
+
+    public void track(Event event, EventOptions options) {
+        this.track(event, options, null);
+    }
+
+    public void track(Event event, EventOptions options, Map<String, Object> extra) {
+        if (!this.isInitializedAndEnabled()) {
+            return;
+        }
+        this.handleEventOptions(options, null);
+        this.client.logEvent(event.eventType, this.getEventPropertiesJson(event));
+    }
+
+    public void identify(String userId, IdentifyProperties properties) {
+        this.identify(userId, properties, null, null);
+    }
+
+    public void identify(String userId, IdentifyProperties properties, EventOptions options) {
+        this.identify(userId, properties, options, null);
+    }
+
+    public void identify(String userId, IdentifyProperties properties, EventOptions options, Map<String, Object> extra) {
```

Comment:
Extra will be a JSONObject
https://github.com/amplitude/Amplitude-Android/pull/308/files#diff-f9ec4ad04ab6ea07630bc809a1f933661cb3dc4423e4d4ce07752f5128f8df0eR5

Reply:
Will fix with middleware support

Reply:
Fixed